plexus-compilers/plexus-compiler-aspectj/src/main/java/org/codehaus/plexus/compiler/ajc/AspectJCompiler.java

```diff
@@ -295,7 +300,7 @@
 
     public AspectJCompiler()
     {
-        super( CompilerOutputStyle.ONE_OUTPUT_FILE_PER_INPUT_FILE, ".java", ".class", null );
+        super( CompilerOutputStyle.ONE_OUTPUT_FILE_PER_INPUT_FILE, "", ".class", null );
```

Comment:
I assume this is a hack to support `.aj` and `.java`. Should be added as a comment, because without that context this line doesn't make any sense.

Reply:
What you call a "hack" is the only way to achieve the desired behaviour. I am just doing what Groovy-Eclipse does, too. The alternative would be to extens Maven Compiler to support a set of source file extensions. I even briefly looked into that, but it would have meant to
  * add an opverloaded constructor,
  * keep the getter for the single value for backward compatibility,
  * add another getter for a set of values,
  * handling the case that the single extension getter is called in case multiple extensions exist (throw an exception? return the first value? what is the first value in a set? do we need a unique list instead?).

I wanted to keep it simple and did this. See lines 663 through 668.

I am going to add a source code comment. I just wanted to get the "hack" argument off the table first. 🙂